airflow/models.py

```diff
@@ -212,33 +212,34 @@ def kill_zombies(self, session):
         Fails tasks that haven't had a heartbeat in too long
         """
         from airflow.jobs import LocalTaskJob as LJ
+        TI = TaskInstance
         logging.info("Finding 'running' jobs without a recent heartbeat")
         secs = (
             configuration.getint('scheduler', 'job_heartbeat_sec') * 3) + 120
         limit_dttm = datetime.now() - timedelta(seconds=secs)
         logging.info(
             "Failing jobs without heartbeat after {}".format(limit_dttm))
-        jobs = (
-            session
-            .query(LJ)
+
+        tis = (
```

Comment:
Shouldn't the base table still be the job? Or I guess in a way I feel that we need to kill the full outer join of this : 
Case 1) Zombie task instance TI.state == State.Running and LJ.State != State.Running and 
         2) LJ.State == State.Running and TI.state != State.Running.

Still need to think about some of this.